packages/core/src/indexing-store/historical.ts

```diff
@@ -687,52 +334,27 @@ export const createHistoricalIndexingStore = ({
     // @ts-ignore
     delete: (table: Table, key) =>
       queue.add(() =>
-        database.wrap(
-          { method: `${getTableName(table) ?? "unknown"}.delete()` },
+        database.record(
+          {
+            method: `${getTableName(table) ?? "unknown"}.delete()`,
+          },
           async () => {
             checkOnchainTable(table, "delete");
-
-            const entry = getCacheEntry(table, key);
-            deleteCacheEntry(table, key);
-
-            if (entry?.row) {
-              if (entry.type === EntryType.INSERT) {
-                return true;
-              }
-
-              await database.qb.drizzle
-                .delete(table)
-                .where(getWhereCondition(table, key));
-
-              return true;
-            } else {
-              if (isDatabaseEmpty) {
-                return false;
-              }
-
-              const deleteResult = await database.qb.drizzle
-                .delete(table as Table)
-                .where(getWhereCondition(table as Table, key))
-                .returning();
-
-              return deleteResult.length > 0;
-            }
+            return indexingCache.delete({ table, key, db });
           },
         ),
       ),
     // @ts-ignore
     sql: drizzle(
       async (_sql, params, method, typings) => {
-        await database.createTriggers();
-        await indexingStore.flush();
-        await database.removeTriggers();
-        isDatabaseEmpty = false;
+        await indexingCache.flush({ db });
```

Comment:
Do we need to call cache.commit here?

Reply:
No, we only want to call cache.commit when we are sure that we will never have to retry that batch of events